proposals/TP1-xds-transport-next.md

```diff
@@ -0,0 +1,1028 @@
+state-of-the-world updates. This is achieved by sending the entire state, e.g.
+every listener inlined in a `ListenerCollection`.
+
+In both the xDS transport and resources, existing fields for managing resource
+names will continue to be used. A client and server will detect `xdstp://`
+semantics by examining the prefix of a resource string.
+
+Clients are expected to have knowledge ahead of time (via mechanisms not part of
+this proposal, e.g. support documentation) on whether the servers specified in
+the bootstrap supports the new URI conventions.
+
+The server may issue redirects by including an `xds::core::v3::ResourceLocator`
+in a `Resource` wrapped in the `Any` message. Redirects are only supported for
+list collections. Glob collections may have individual resources redirected but the
+collection itself is not redirectable.
+
+### Examples
+
+We provide YAML examples below of discovery request/response sequences for various use cases.
+
+#### Singleton resource request
+
+Client `Cluster` resource:
+
+
+```
+name: some-cluster
+eds_cluster_config:
+  service_name: xdstp://some-authority/envoy.config.endpoint.v3.ClusterLoadAssignment/foo
+  eds_config:
+    authorities:
+    - name: some-authority
+  … rest of ConfigSource pointing at xDS management server … 
+```
+
+Client EDS `DeltaDiscoveryRequest` sent to xDS management server:
+
+
+```
+resource_names_subscribe:
+- xdstp://some-authority/envoy.config.endpoint.v3.ClusterLoadAssignment/foo
+```
+
+
+xDS management server `DeltaDiscoveryResponse` sent to client:
+
+
+```
+resources:
+- version: 1
+  name: xdstp://some-authority/envoy.config.endpoint.v3.ClusterLoadAssignment/foo
+  resource:
+    "@type": type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment
+    … foo's ClusterLoadAssignment payload … 
+```
+
+### Collection resource requests
+
+#### List collections 
+
+Client bootstrap:
+
+
+```
+dynamic_resources:
+  lds_resources_locator:
+    name: xdstp://some-authority/envoy.config.listeners.v3.ListenerCollection/foo
+  lds_config:
+    authorities:
+    - name: some-authority
+  … rest of ConfigSource pointing at xDS management server … 
+```
+
+
+Client LDS `DeltaDiscoveryRequest` sent to xDS management server:
+
+
+```
+resource_names_subscribe:
+- xdstp://some-authority/envoy.config.listeners.v3.ListenerCollection/foo
+```
+
+
+xDS management server `DeltaDiscoveryResponse` sent to client:
+
+
+```
+resources:
+- version: 1
+  name: xdstp://some-authority/envoy.config.listeners.v3.ListenerCollection/foo
+  resource:
+    "@type": type.googleapis.com/envoy.config.listeners.v3.ListenerCollection
+    - locator: xdstp://some-authority/envoy.config.listeners.v3.Listener/bar
+    - locator: xdstp://some-authority/envoy.config.listeners.v3.Listener/baz
+```
+
+
+The `bar` and `baz` resources are then fetched as per [singleton resource
+requests](#singleton-resource-request).
+
+#### List collections with inlining
```

Comment:
How would server know whether to include inline resources or not?

Reply:
I added a comment below, but basically this is at the discretion of the server and it decides based on criteria such as reducing round-trips.